core/src/test/java/io/confluent/rest/Http2Test.java

```diff
@@ -0,0 +1,340 @@
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * This test is a bit unusual because of the way that multiple Java versions are handled.
+ * Essentially, the HTTP/2 support is only properly viable in Java 9 and later. Actually,
+ * Java 11 is the first LTS version after that. Because we do not want to have a variable
+ * set of dependencies based on which version of Java was used to build and package,
+ * this test is packaged to exploit HTTP/2 on Java 11 and then conditionally avoid that
+ * code on earlier versions of Java. This is the client equivalent of how the HTTP/2
+ * support works in the server.
+ **/
+public class Http2Test {
+  private static final Logger log = LoggerFactory.getLogger(Http2Test.class);
+
+  private File trustStore;
+  private File clientKeystore;
+  private File serverKeystore;
+
+  private static final String HTTP_URI = "http://localhost:8080";
+  private static final String HTTPS_URI = "https://localhost:8081";
+  private static final String SSL_PASSWORD = "test1234";
+  private static final String EXPECTED_200_MSG = "Response status must be 200.";
+
+  @Before
+  public void setUp() throws Exception {
+    try {
+      trustStore = File.createTempFile("Http2Test-truststore", ".jks");
+      clientKeystore = File.createTempFile("Http2Test-client-keystore", ".jks");
+      serverKeystore = File.createTempFile("Http2Test-server-keystore", ".jks");
+    } catch (IOException ioe) {
+      throw new RuntimeException("Unable to create temporary files for trust stores and keystores.");
+    }
+    Map<String, X509Certificate> certs = new HashMap<>();
+    createKeystoreWithCert(clientKeystore, "client", certs);
+    createKeystoreWithCert(serverKeystore, "server", certs);
+    TestSslUtils.createTrustStore(trustStore.getAbsolutePath(), new Password(SSL_PASSWORD), certs);
+
+    TestMetricsReporter.reset();
+  }
+
+  private void createKeystoreWithCert(File file, String alias, Map<String, X509Certificate> certs) throws Exception {
+    KeyPair keypair = TestSslUtils.generateKeyPair("RSA");
+    CertificateBuilder certificateBuilder = new CertificateBuilder(30, "SHA1withRSA");
+    X509Certificate cCert = certificateBuilder.sanDnsNames("localhost")
+        .generate("CN=mymachine.local, O=A client", keypair);
+    TestSslUtils.createKeyStore(file.getPath(), new Password(SSL_PASSWORD), new Password(SSL_PASSWORD),alias, keypair.getPrivate(), cCert);
+    certs.put(alias, cCert);
+  }
+
+  private void configServerKeystore(Properties props) {
+    props.put(RestConfig.SSL_KEYSTORE_LOCATION_CONFIG, serverKeystore.getAbsolutePath());
+    props.put(RestConfig.SSL_KEYSTORE_PASSWORD_CONFIG, SSL_PASSWORD);
+    props.put(RestConfig.SSL_KEY_PASSWORD_CONFIG, SSL_PASSWORD);
+  }
+
+  private TestRestConfig buildTestConfig(boolean httpListener,
+                                         boolean httpsListener,
+                                        boolean disableHttp2) {
```

Comment:
nit: Indent looks wrong here. Also I would prefer having this flag be `enableHttp2` (the important part being it being `true` if HTTP/2 is enabled, not the actual name) , as this is how we define the flag in `Application.java`.